### PR TITLE
[RDBMS] `az postgres flexible-server replica`: Remove stop-replication command support

### PR DESCRIPTION
**Pull request by nasc17**
**Related command**
`az postgres flexible-server replica`

**Description**
Dec 2023: “promote” was added to be used instead of “stop-replication”: Stop replication is the term used in single server. Was inherited to flex but needed to be updated to promote.
Jan 2024: was hidden and deprecation warning was set

**Testing Guide**
azdev test

**History Notes**
[RDBMS] `az postgres flexible-server replica`: Remove stop-replication command support

### CODE REVIEW

**Comment by nasc17** [comment]
@evelyn-ys please trigger checks and review.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py [hunk 1]
```diff
@@ -238,7 +238,6 @@ def load_flexibleserver_command_table(self, _):
                             custom_command_type=flexible_servers_custom_postgres,
                             client_factory=cf_postgres_flexible_servers) as g:
         g.custom_command('create', 'flexible_replica_create', supports_no_wait=True)
-        g.custom_command('stop-replication', 'flexible_replica_stop', confirmation=True, deprecate_info=g.deprecate(redirect='postgres flexible-server replica promote', hide=True))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:241]: This is breaking change, can't be merged until next breaking change window

**Comment by nasc17** [comment]
Need to wait for next breaking change window

**Pull request closed without merging**